### ISSUE TITLE
Updated `ColorPicker`

### ISSUE THREAD

**Issue by ciampo** [WordPress/gutenberg#34284]
_This is an overview issue to track all of the tasks necessary to update the `ColorPicker` component with the new design and primitives._

![image](https://user-images.githubusercontent.com/548849/130771534-3cba93c1-d33f-40dd-967c-8343703b5240.png)

### Component

- [x] Add experimental `ColorPicker` component to the repo (done in https://github.com/WordPress/gutenberg/pull/33714)
- [x] #34286
- [x] #34289
- [x] #34795
- [x] Match design proposal: (#34598)
  - [x] Adjust the white border on the circle controls to match design (compare storybook with design mockup).
  - [x] The input padding is off, not sure if this is something we need to affect on all inputs or we can just start with the input + slider case.
  - [x] Spacing between slider and input is off.
  - [x] #34288
  - [x] Use the new settings icon from wp/icons.
  - [x] the color format dropdown should be rendered without borders (not sure how we want to handle this at the component level since it should be an optional thing used rarely)
  - [x] Adjust circular handle of slider since it's too big compared to designs
- [ ] Follow-up tasks following previous design enhancement PR:
  - [ ] Redesign the input fields taking into consideration the new border color and height (removing specific height from this component).
  - [ ] Improve the RangeControl making the focus work on safari and implementing the new focus design.
  - [ ] Address the glitch with circle borders/shadows on chrome.
- [ ] #34487
- [x] #35675
- [x] #35677
- [ ] #38065
- [ ] #38066

### Integration

- [x] Add deprecated props adapter to transform the props for the "legacy" `ColorPicker` into the new format (done in https://github.com/WordPress/gutenberg/pull/34014)
- [x] #34287
- [x] Enable Alpha and close #18095

### Related issues

I've also found a list of open issues/PRs that will likely be closed (or should otherwise be fixed) as a consequence of the work carried out in the tasks above:

- [x] #19785
- [x] #29544
- [ ] #28758
- [x] #26568
- [ ] #22890 (with open PR #26770)
- [x] #36314
- [ ] #11791

**Comment by mtias** [comment]
@ciampo are the open tasks still being actively pursued? Do they need to be tracked here or can this be considered done?

**Comment by ciampo** [comment]
This issue should be quite up-to-date. The components squad hasn't quite had the capacity to work on these tasks (to the point that the latest work re. the `ColorPicker` was done with the help of @jorgefilipecosta ), but it'd be great if we could work the final few tasks left.

**Comment by andreiglingeanu** [comment]
@ciampo Is there any way to help with this so that we can get this moving ahead? Can help with providing fixes for the remaining stuff.

**Comment by Mamaduka** [comment]
@ciampo, can we close #22890 and #26770? I think refactoring is no longer needed after #33714.

**Comment by ciampo** [comment]
Hey @Mamaduka ! 

- https://github.com/WordPress/gutenberg/pull/26770 can be closed, as you pointed out
- https://github.com/WordPress/gutenberg/issues/22890 should be kept open, since there are still a few React Class components in the `@wordpress/components` package (`ColorPicker` was only one of them)

**Comment by Mamaduka** [comment]
Thanks, @ciampo!

**Comment by annezazu** [comment]
Doing some clean up and closing this out as it seems most initiatives have been handled and active work hasn't continued in a few years. If this is incorrect, happy to reopen!
